test/e2e/utils/pod.go

```diff
@@ -221,6 +237,12 @@ func nfdWorkerPodSpec(image string, extraArgs []string) *v1.PodSpec {
 						},
 					},
 				},
+				SecurityContext: &v1.SecurityContext{
+					AllowPrivilegeEscalation: &allowPrivilegeEscalation,
```

Comment:
I think you could also add `readOnlyRootFilesystem`
```suggestion
					AllowPrivilegeEscalation: &allowPrivilegeEscalation,
					readOnlyRootFilesystem: true
```